src/main/java/org/sonarsource/sonarlint/ls/connected/events/ServerSentEventsHandler.java

```diff
@@ -55,9 +64,24 @@ public void handleEvents(ServerEvent event) {
       event instanceof IssueChangedEvent) {
       taintVulnerabilitiesCache.getAllFilesWithTaintIssues()
         .forEach(projectBindingManager::updateTaintIssueCacheFromStorageForFile);
+    } else if (event instanceof SecurityHotspotRaisedEvent || event instanceof SecurityHotspotChangedEvent || event instanceof SecurityHotspotClosedEvent) {
+      var fileUri = getFileUriFromEvent(event);
+      fileUri.ifPresent(analysisScheduler::didReceiveHotspotEvent);
     }
   }
 
+  private Optional<URI> getFileUriFromEvent(ServerEvent event) {
+    var serverPath = "";
+    if (event instanceof SecurityHotspotRaisedEvent) {
```

Comment:
You could create in the core the common interface for the `HotspotServerEvent` that has this `getFilePath()` method. It would spare you from multiple instanceof checks and boilerplate code below.